File: Swift/ExampleTests/common/ApiRequestTyped.swift
public struct ApiRequestTyped: Equatable, CustomStringConvertible {
    public let method: String
    public let page: String
    public let parameter: String
    public let body: String

    public init(method: String, page: String, parameter: String, body: String) {
        self.method = method
        self.page = page
        self.parameter = parameter
        self.body = body
    }

    public init(from s: ApiRequestString) {
        self.method = s.method
        self.page = s.page
        self.parameter = s.parameter
        self.body = s.body
    }

    public func toStringStruct() -> ApiRequestString {
        return ApiRequestString(
            method: String(describing: method),
            page: String(describing: page),
            parameter: String(describing: parameter),
            body: String(describing: body)
        )
    }

    public static func toStringList(_ list: [ApiRequestTyped]) -> [ApiRequestString] {
        return list.map { $0.toStringStruct() }
    }

    public static func fromStringList(_ list: [ApiRequestString]) -> [ApiRequestTyped] {
        return list.map { ApiRequestTyped(from: $0) }
    }

    public func toJSONValue() -> [String: Any] {
        return [
            "method": method,
            "page": page,
            "parameter": parameter,
            "body": body,
        ]
    }

    public func toJSON() throws -> String {
        return try Json.write(toJSONValue())
    }

    public init(fromJSONValue m: [String: Any]) throws {
        self.method = try Json.asString(Json.require(m, "method"), "method")
        self.page = try Json.asString(Json.require(m, "page"), "page")
        self.parameter = try Json.asString(Json.require(m, "parameter"), "parameter")
        self.body = try Json.asString(Json.require(m, "body"), "body")
    }

    public init(fromJSON text: String) throws {
        try self.init(fromJSONValue: Json.parseObject(text))
    }

    public static func toJSONList(_ list: [ApiRequestTyped]) throws -> String {
        return try Json.write(list.map { $0.toJSONValue() })
    }

    public static func fromJSONList(_ text: String) throws -> [ApiRequestTyped] {
        return try Json.parseArray(text).map {
            try ApiRequestTyped(fromJSONValue: Json.asObject($0, "ApiRequestTyped"))
        }
    }

    public var description: String {
        return "Method=\(method), Page=\(page), Parameter=\(parameter), Body=\(body)"
    }
}

File: Rust/ExampleTests/common/apirequest_typed.rs
#![allow(dead_code, unused_imports, unused_variables)]

use super::json;
use super::apirequest_string::ApiRequestString;

#[derive(Debug, Clone, Default, PartialEq)]
pub struct ApiRequestTyped {
    pub method: String,
    pub page: String,
    pub parameter: String,
    pub body: String,
}

impl ApiRequestTyped {
    pub fn from_str_struct(s: &ApiRequestString) -> Self {
        Self {
            method: s.method.clone(),
            page: s.page.clone(),
            parameter: s.parameter.clone(),
            body: s.body.clone(),
        }
    }

    pub fn to_str_struct(&self) -> ApiRequestString {
        ApiRequestString {
            method: self.method.clone(),
            page: self.page.clone(),
            parameter: self.parameter.clone(),
            body: self.body.clone(),
        }
    }

    pub fn to_string_list(list: &[ApiRequestTyped]) -> Vec<ApiRequestString> {
        list.iter().map(|t| t.to_str_struct()).collect()
    }

    pub fn from_string_list(list: &[ApiRequestString]) -> Vec<ApiRequestTyped> {
        list.iter().map(ApiRequestTyped::from_str_struct).collect()
    }

    pub fn to_json_value(&self) -> json::Value {
        json::Value::Object(vec![
            ("Method".to_string(), json::Value::Str(self.method.clone())),
            ("Page".to_string(), json::Value::Str(self.page.clone())),
            ("Parameter".to_string(), json::Value::Str(self.parameter.clone())),
            ("Body".to_string(), json::Value::Str(self.body.clone())),
        ])
    }

    pub fn to_json(&self) -> String {
        json::write(&self.to_json_value())
    }

    pub fn from_json_value(v: &json::Value) -> json::JsonResult<Self> {
        Ok(Self {
            method: json::as_string(json::require(v, "Method")?, "Method")?,
            page: json::as_string(json::require(v, "Page")?, "Page")?,
            parameter: json::as_string(json::require(v, "Parameter")?, "Parameter")?,
            body: json::as_string(json::require(v, "Body")?, "Body")?,
        })
    }

    pub fn from_json(text: &str) -> json::JsonResult<Self> {
        Self::from_json_value(&json::parse(text)?)
    }

    pub fn to_json_list(list: &[ApiRequestTyped]) -> String {
        json::write(&json::Value::Array(
            list.iter().map(|item| item.to_json_value()).collect(),
        ))
    }

    pub fn from_json_list(text: &str) -> json::JsonResult<Vec<ApiRequestTyped>> {
        let root = json::parse(text)?;
        let items = json::as_array(&root, "ApiRequestTyped")?;
        let mut result = Vec::with_capacity(items.len());
        for e in items {
            result.push(Self::from_json_value(e)?);
        }
        Ok(result)
    }
}
